src/dashboard/src/media/js/archival_storage/archival_storage_search.js

```diff
@@ -132,6 +132,13 @@ function renderArchivalStorageSearchForm(search_uri, on_success, on_error) {
 }
 
 $(document).ready(function() {
+  // If user arrived via browser back or forward button, reload page to prevent
+  // a DataTables error in Chrome that results from inconsistencies between the
+  // cached DOM and the data returned by the AJAX call.
+  var backOrForwardNavigation = navigatedFromBackOrForwardButton();
+  if (backOrForwardNavigation === true) {
+    window.location.reload();
+  }
```

Comment:
Thanks @ross-spencer! I share your gut feeling that the page reload is a bit of a blunt force approach, and appreciate your proposal. While trying to implement something like what you suggest, I realized that every time the user hits back in Chrome from the AIP details page, `showFiles` is always evaluated as `false` on line `var showFiles = $('#id_show_files').prop('checked');`, regardless of the status of the checkbox. Hence why your code to uncheck the "show files" checkbox isn't working - the conditions to satisfy the if statement around it are never met. Whereas in Firefox, `showFiles` appears to be evaluated correctly based on whether or not the "show files" checkbox is checked.

I think I might have figured out a way around this. Will keep investigating and update when I have something more definitive.

Reply:
Interestingly, just setting `var showFiles = $('#id_show_files').prop('checked');` and then using the `showFiles` constant through the rest of the `get_datatable` function is enough to resolve the DataTables error. But results in new UI inconsistencies.

Reply:
Note for self: once `navigatedFromBackOrForwardButton` returns true, it continues to return true until the page is reloaded or the user clicks through to a new page. In practice, this means adding something like this to the start of `get_datatable()`:

```
backOrForwardNavigation = navigatedFromBackOrForwardButton();
if (backOrForwardNavigation === true) {
  showFiles = false;
  $('#id_show_files').prop('checked', false);
}
```
prevents the "show files" checkbox from working as expected until the page is reloaded or the user goes to another page and then returns to the Archival Storage tab by a means other than their browser's back or forward buttons.

Reply:
Interesting blog post related to this issue: https://madhatted.com/2013/6/16/you-do-not-understand-browser-history

The following stood out to me as a potentially relevant passage for understanding what's going on in this issue:

"Safari and Firefox implement something we can call the 'bfcache' (back-forward cache). This cache stores the current state of the DOM in memory. When a page is re-visited via the back button, no requests are made. The DOM itself is served straight up from cache."

So we're not getting the issue in Firefox because Firefox is simply returning the DOM from its cache, whereas Chrome is attempting to build the page from scratch (though seemingly in a way that is persisting the "show files" checkbox state in a way that's problematic for us).

Interestingly, it looks like Google added an experimental back-forward cache in Chrome 79, just a few months ago: https://www.zdnet.com/article/chrome-79-released-with-tab-freezing-back-forward-caching-and-loads-of-security-features/

Reply:
It turns out that the checkbox state was being cached by `autocomplete` in the Archival Storage search form! Pushing a much simpler fix that brings consistency between browsers in just a minute.